### PR TITLE
Github actions migration

### PR DESCRIPTION
**Pull request by paualarco**
This PR aims to migrate from Travis to GitHub Actions.

The workflow has been divided between 5 jobs: 
- Runs JVM tests: These are currently failing and yet don't know why, see [error log].(https://github.com/paualarco/monix-bio/runs/705430137?check_suite_focus=true).
- JS tests: Runs Having a separated test for JS allows to make JVM and JS to run in parallel.
- Mima binary compatibility test
- Publish: This job depends on the success of the previous ones jvm, js and mima), not sure how this project was doing releases but it has been configured as same as `monix-connect`, therefore it is required to configure some secrets in the project secrets, see [sbt-ci-release](https://github.com/olafurpg/sbt-ci-release).
- Publish microsite: It depends on publish job.

### CODE REVIEW

**Comment by Avasil** [comment]
> Runs JVM tests: These are currently failing and yet don't know why, see [error log].

It seems like unidoc can't resolve links to `monix/monix`. I've had issues with it before, I'll see if I can solve it again

> Publish: This job depends on the success of the previous ones jvm, js and mima), not sure how this project was doing releases but it has been configured as same as monix-connect, therefore it is required to configure some secrets in the project secrets, see sbt-ci-release.

I was doing releases manually 😅 but it would be awesome to make it automated!

File: .github/scripts/build [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    exit 1
+fi
+
+if [[ -z "$SBT_COMMAND" ]]; then
+    echo "ERROR: environment value SBT_COMMAND is not set!" 1>&2
+    exit 1
+fi
+
+
+if [ $SCALA_VERSION == '2.11.12' ] && [ $SCALAJS_VERSION == '1.0.0' ]; then
```
Comment (Avasil) [inline .github/scripts/build:27]: I don't use 2.11 in `monix-bio` so this if can be removed

Comment (paualarco) [reply .github/scripts/build]: removed

**Comment by paualarco** [comment]
> It seems like unidoc can't resolve links to `monix/monix`. I've had issues with it before, I'll see if I can solve it again

The pipeline looks fine now :D 
Although you might want to add something else.
So, since you would need to set up [some secrets](https://github.com/olafurpg/sbt-ci-release) for automated release, being `PGP_PASSPHRASE`, `PGP_SECRET`, `SONATYPE_PASSWORD` and `SONATYPE_USER`.
Finally for publishing page, you would just have to add the secret key `GIT_DEPLOY_KEY` as you did already for travis.
Notice that for travis it started with `GITHUB_`, but this prefix is reserved in Github.

**Comment by Avasil** [comment]
Thank you, I will try it out today :)

**Comment by alexandru** [comment]
I like the separation between JVM/JS/Mima tests.

Could you contribute a setup like this to monix/monix too? 🙂

One note — the deployment step could be a separate workflow and could be triggered in response to marking a tag as being a "release". Not sure we want to do this, I guess when triggering a release, an extra round of testing would be nice. But the explicit marking of a tag as being a "release" is useful imo to avoid mistakes.

```yaml
name: deploy
on:
  release:
    types: [released]

# ...
```

**Comment by alexandru** [comment]
One more improvement suggestion — it appears you haven't configured caching and this can improve the build times:

```yaml
      - name: Cache SBT / Travis
        uses: actions/cache@v2
        with:
          path: ~/.cache/coursier
          key: ${{ runner.os }}-sbt-coursier-${{ hashFiles('**/build.sbt') }}
```

(NOTE here — that path is only available on Linux, if we'll ever want to configure GitHub Actions to run our tests with Windows or MacOS, we'll need to setup platform specific paths)

**Comment by Avasil** [comment]
> Could you contribute a setup like this to monix/monix too? 🙂

There is PR: https://github.com/monix/monix/pull/1189

@paualarco I had to do small tweaks related to crossproject release but it seems to be working! Though I'm trying to include a badge with the latest version but unfortunately it is just returning old `0.0.2-SNAPSHOT`

**Comment by alexandru** [comment]
> There is PR: monix/monix#1189

Ah, awesome 👍

**Comment by paualarco** [comment]
> One more improvement suggestion — it appears you haven't configured caching and this can improve the build times:

That's good to know :)

> Could you contribute a setup like this to monix/monix too? 🙂

That one was the first I did 😆 
Indeed I took advantage of the work that you started.
I can create a PR for the rest of the monix projects as well :)

**Pull request merged**